File: testapp/src/main/java/com/juanimoli/strictmodenotifier/testapp/CustomLogWatchService.java
package com.juanimoli.strictmodenotifier.testapp;

import android.util.Log;

import com.juanimoli.strictmodenotifier.LogWatchService;
import com.juanimoli.strictmodenotifier.commons.StrictModeViolation;

public class CustomLogWatchService extends LogWatchService {

    @Override
    protected void notifyViolation(StrictModeViolation violation) {
        super.notifyViolation(violation);

        //Custom Actions.
        // ex) Send logs to Slack.
        Log.d("CustomLogWatchService", "custom action");
    }
}

File: testapp/src/main/java/io/github/juanimoli/strictmode/notifier/service/CustomLogWatchService.kt
package io.github.juanimoli.strictmode.notifier.service

import android.util.Log
import io.github.juanimoli.strictmode.notifier.LogWatchService
import io.github.juanimoli.strictmode.notifier.commons.StrictModeViolation

class CustomLogWatchService : LogWatchService() {
    override fun notifyViolation(violation: StrictModeViolation) {
        super.notifyViolation(violation)

        //Custom Actions.
        // ex) Send logs to Slack.
        Log.d("CustomLogWatchService", "custom action")
    }
}
